File: 1823.找出游戏的获胜者.go
/*
 * @lc app=leetcode.cn id=1823 lang=golang
 *
 * [1823] 找出游戏的获胜者
 */

// @lc code=start
func findTheWinner(n, k int) int {
	q := make([]int, n)
	for i := range q {
		q[i] = i + 1
	}
	for len(q) > 1 {
		for i := 1; i < k; i++ {
			q = append(q, q[0])[1:]
		}
		q = q[1:]
	}
	return q[0]
}

// @lc code=end

File: 1823.找出游戏的获胜者.py
#
# @lc app=leetcode.cn id=1823 lang=python3
#
# [1823] 找出游戏的获胜者
#

# @lc code=start
class Solution:
    def findTheWinner(self, n: int, k: int) -> int:
        q = deque(range(1, n + 1))
        while len(q) > 1:
            for _ in range(k - 1):
                q.append(q.popleft())
            q.popleft()
        return q[0]
# @lc code=end

